mockwebserver/src/main/kotlin/mockwebserver3/MockResponse.kt

```diff
@@ -152,7 +148,37 @@ class MockResponse {
 
     internal var trailers: Headers.Builder
 
+    // At most one of (body,webSocketListener,streamHandler) is non-null.
+    private var body_: MockResponseBody? = null
+    private var streamHandler_: StreamHandler? = null
+    private var webSocketListener_: WebSocketListener? = null
     var body: MockResponseBody?
+      get() = body_
+      set(value) {
+        body_ = value
+        if (value != null) {
+          streamHandler_ = null
+          webSocketListener_ = null
+        }
+      }
+    var streamHandler: StreamHandler?
+      get() = streamHandler_
+      set(value) {
+        streamHandler_ = value
+        if (value != null) {
+          body_ = null
+          webSocketListener_ = null
+        }
+      }
+    var webSocketListener: WebSocketListener?
+      get() = webSocketListener_
+      set(value) {
+        webSocketListener_ = value
+        if (value != null) {
```

Comment:
This seems confusing. Specifically a null not clearing the other ones.

Might be better to fail if this is non null and other fields also? Make them explicitly clear other fields first?

But your call.

Reply:
Yeah, I was trying to avoid awkwardness when doing a manual copy:

```
val a: MockResponse = ...
val b = MockResponse.Builder()
  .body(a.body)
  .webSocketListener(a.webSocketListener)
  .streams(a.streams)
```

I’ve probably overthunk it. I’ll change to make null not-special.




```